testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2290,6 +2290,8 @@ public void testCreateTableAsSelect()
                         "SELECT 1234567890, 1.23",
                 "SELECT count(*) + 1 FROM customer");
 
+        // TODO: BigQuery throws table not found at BigQueryClient.insert if we reuse the same table name
```

Comment:
Can you explain a bit more to me? Does this mean that when the DROP TABLE above runs and then we try to CREATE TABLE with same name again here it fails?

Reply:
> Does this mean that when the DROP TABLE above runs and then we try to CREATE TABLE with same name again here it fails?

Yes. Also, the issue doesn't happen in case of empty tables. It seems there's cache or something for `BigQuery.insertAll`. I couldn't find relevant options in the library at glance. I will look into the detail later.